prov/gni/src/gnix_wait.c

```diff
@@ -195,17 +344,13 @@ static int gnix_init_wait_obj(struct gnix_fid_wait *wait, enum fi_wait_obj type)
 	wait->type = type;
 
 	switch (type) {
-	case FI_WAIT_FD:
+	case FI_WAIT_UNSPEC:
 		if (socketpair(AF_LOCAL, SOCK_STREAM, 0, wait->fd))
 			goto err;
 
 		if (fi_fd_nonblock(wait->fd[WAIT_READ]))
 			goto cleanup;
 		break;
-	case FI_WAIT_MUTEX_COND:
-		pthread_mutex_init(&wait->mutex, NULL);
-		pthread_cond_init(&wait->cond, NULL);
-		break;
```

Comment:
We don't support FI_WAIT_MUTEX_COND currently, so the code was removed, I can leave it in as unsused code that we could use in the future if we end up supporting directly using FI_WAIT_FD / FI_WAIT_COND in the future if people deem that something they wish done.